backend/Controllers/UserController.cs

```diff
@@ -21,4 +21,31 @@ public async Task<IActionResult> GetByUserName([FromRoute] string userName)
         }
         return Ok(new UserResponse { Id = matchingUser.Id, UserName = matchingUser.UserName!, });
     }
+
+    [HttpDelete("{id}")]
+    public async Task<IActionResult> Delete([FromRoute] string id)
+    {
+        var user = await _userManager.FindByIdAsync(id);
+
+        if (user == null)
+        {
+            return NotFound();
+        }
+
+        var result = await _userManager.DeleteAsync(user);
```

Comment:
accessing the database for the delete should be happening in the user service not the controller.
line 28 should also be happening in the user service not the controller - I can see the endpoint above is doing the same thing in the controller (I didn't catch that in a previous review) please could you add methods into the User Service to make each of these database calls and amend the controller to call the?

Reply:
All done :)